src/blockchain/process.rs

```diff
@@ -1,14 +1,15 @@
-use crate::blockcfg::Header;
 use crate::blockchain::chain::{self, BlockHeaderTriage, HandledBlock};
-use crate::intercom::BlockMsg;
+use crate::intercom::{BlockMsg, NetworkPropagateMsg};
 use crate::rest::v0::node::stats::StatsCounter;
-use crate::utils::task::TaskBroadcastBox;
+
 use chain_core::property::Header as _;
 
+use futures::sync::mpsc;
+
 pub fn process(
     blockchain: &chain::BlockchainR,
     bquery: BlockMsg,
-    network_broadcast: &mut TaskBroadcastBox<Header>,
```

Comment:
can you hid this behind the TaskBroadcastBox instead then ? It is the only object utilsign the TaskboardcastBox and we introduced abstraction to not have to change the whole code everytime we decide to change how it is working internally

Reply:
`TaskMessageBox` then, as it won't be a broadcast message any more.

Reply:
Trouble is, `TaskMessageBox` can only be constructed by `Services::spawn_with_inputs`. But we have a chicken and egg problem here, as the network task is constructed last and the block task would need its message box to send broadcasts. Previously, there was also an `UnboundedSender` there which I restored from history; I guess it was done with the abstraction-breaking workaround for the same reason.

Reply:
I should think of adding another abstraction type to `network::propagate` that will encapsulate the sender.

Reply:
commit dc6ec7cbdd8cfa77510e157ae5c5d4cd025fdd43 provides an abstraction.